This PR fixes compiling on Ubuntu 22.04 arm64, but breaks compiling on macOS 13.0.1 arm64:

```
[ 43%] Compiling thirdparty/embree/common/sys/sysinfo.cpp ...
In file included from thirdparty/embree/common/sys/sysinfo.cpp:5:
In file included from thirdparty/embree/common/sys/intrinsics.h:13:
In file included from thirdparty/embree/common/sys/../simd/arm/emulation.h:12:
thirdparty/embree/common/sys/../simd/arm/sse2neon.h:5957:33: error: cannot initialize a parameter of type 'float32x4_t' (vector of 4 'float32_t' values) with an lvalue of type '__m128d' (aka 'float64x2_t')
    __builtin_nontemporal_store(a, (float32x4_t *) p);
                                ^
```

It seems really strange that this configuration is required on Linux but breaks everywhere else...

--- a/modules/raycast/SCsub
+++ b/modules/raycast/SCsub
@@ -68,10 +68,10 @@ if env["builtin_embree"]:
 
     if not env.msvc:
         if env["arch"] in ["x86_64", "x86_32"]:
-            env_raycast.Append(CPPFLAGS=["-msse2", "-mxsave"])
+            env_raycast.Append(CCFLAGS=["-msse2", "-mxsave"])
 
         if env["platform"] == "windows":
-            env_raycast.Append(CPPFLAGS=["-mstackrealign"])
+            env_raycast.Append(CCFLAGS=["-mstackrealign"])
 
     if env["platform"] == "windows":
         if env.msvc:
@@ -92,11 +92,15 @@ if env["builtin_embree"]:
         env_thirdparty.Append(CPPDEFINES=["__SSE2__", "__SSE__"])
 
     if env["platform"] == "web":
-        env_thirdparty.Append(CPPFLAGS=["-msimd128"])
+        env_thirdparty.Append(CXXFLAGS=["-msimd128"])
 
     if not env.msvc:
+        # Flags synced with upstream gnu.cmake.
+        if env["arch"] == "arm64" and env["platform"] == "linuxbsd":
+            env_thirdparty.Append(CXXFLAGS=["-flax-vector-conversions"])
+
         env_thirdparty.Append(
-            CPPFLAGS=[
+            CXXFLAGS=[
                 "-fno-strict-overflow",
                 "-fno-delete-null-pointer-checks",
                 "-fwrapv",

